spring-cloud-stream-module-launcher/src/main/java/org/springframework/cloud/stream/module/utils/ClassloaderUtils.java

```diff
@@ -50,16 +53,23 @@ public static ClassLoader createModuleClassloader(URL[] urls) {
 			// to compensate for LaunchedURLClassLoader not delegating to parent to retrieve resources
 			@SuppressWarnings("resource")
 			URLClassLoader systemUrlClassLoader = (URLClassLoader) systemClassLoader;
-			URL[] mergedUrls = new URL[urls.length + systemUrlClassLoader.getURLs().length];
+			List<URL> systemURLs = new ArrayList<>();
+			for (URL url: systemUrlClassLoader.getURLs()) {
+				// this will help avoiding any tests using the classes directory in the system class loader.
+				if (!url.getFile().endsWith("classes/")) {
```

Comment:
@mbogoevici While I agree we shouldn't be doing exclusions on `hardcoded` path, I thought this specific change makes sense at least in the context of excluding the path that **ends** with `classes` directory in order to remove **non-jar** resources out of the path here (that happens only in case of test run). This is a non-issue on `runtime` environment where we wouldn't have path that ends with `classes/` and I believe all the URLs would end with `jar` or `jar` inside the `exec` jar. What do you think?